packages/react-server/core/renderMiddleware.js

```diff
@@ -736,6 +736,8 @@ function writeBody(req, res, context, start, page) {
 
 		// Let the client know it's not getting any more data.
 		renderScriptsAsync([{ text: `__reactServerClientController.failArrival()` }], res)
+		// Close off the body since the page life cycle will not complete
+		res.write("</div></body></html>");
```

Comment:
I don't think this is enough.  It's addressing the _symptom_ (we're not sending the close tags) rather than the _cause_ (we're short-circuiting out of the normal control flow, which includes sending the close tags).

Can we handle individual element render failures without rejecting the `writeBody` return promise?